### PR TITLE
fix(targets): Quote column names in INSERT statement

### PR DESCRIPTION
**Pull request by mattwill09**
This is an update to fix how the insert into statements are being generated. If you have a source column that uses a reserved word then invalid SQL was being generated. 

Previous:
```sql
INSERT INTO analytics.foo
(id, in)
VALUES (:id, :in)
```

Updated:
```sql
INSERT INTO analytics.foo
("id", "in")
VALUES (:id, :in)
```

----
📚 Documentation preview 📚: https://meltano-sdk--2198.org.readthedocs.build/en/2198/

### CODE REVIEW

File: singer_sdk/sinks/sql.py [hunk 1]
```diff
@@ -282,7 +282,7 @@ def generate_insert_statement(
         statement = dedent(
             f"""\
             INSERT INTO {full_table_name}
-            ({", ".join(property_names)})
+            ({", ".join([f'"{name}"' for name in property_names])})
```
Comment (edgarrmondragon) [inline singer_sdk/sinks/sql.py:285]: We could try using https://docs.sqlalchemy.org/en/20/core/sqlelement.html#sqlalchemy.sql.expression.quoted_name

**Review by edgarrmondragon — CHANGES_REQUESTED** [review]
Thanks @mattwill09!

You may want to update the test in https://github.com/meltano/sdk/blob/af3b678ddda34112b56954d568fe152a7e45dfbf/tests/samples/test_target_sqlite.py#L490-L495.

File: singer_sdk/sinks/sql.py (hunk 1 above)
Comment (mattwill09) [reply singer_sdk/sinks/sql.py:285]: I'm not typically a python developer, but that change looks to be easy enough. I'll make sure to get those tests fixed up too.

File: singer_sdk/sinks/sql.py [hunk 2]
```diff
@@ -282,7 +283,7 @@ def generate_insert_statement(
         statement = dedent(
             f"""\
             INSERT INTO {full_table_name}
-            ({", ".join(property_names)})
+            ({", ".join(quoted_name(name) for name in property_names)})
```
Comment (edgarrmondragon) [inline singer_sdk/sinks/sql.py:286]:
```suggestion
            ({", ".join(quoted_name(name, True) for name in property_names)})
```

**Review by edgarrmondragon — CHANGES_REQUESTED** [review]

File: singer_sdk/sinks/sql.py [hunk 3]
```diff
@@ -282,7 +283,7 @@ def generate_insert_statement(
         statement = dedent(
             f"""\
             INSERT INTO {full_table_name}
-            ({", ".join(property_names)})
+            ({", ".join(quoted_name(name, True) for name in property_names)})
```
Comment (edgarrmondragon) [inline singer_sdk/sinks/sql.py:286]:
```suggestion
            ({", ".join(quoted_name(name, True) for name in property_names)})  # noqa: FBT003
```

**Comment by edgarrmondragon** [comment]
Superseded by #2200. Thanks @mattwill09!

**Comment by mattwill09** [comment]
Thanks for finding a proper solution for this. Much appreciated!

**Pull request closed without merging**